File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.cpp
class Solution {
public:
    vector<int> leftRightDifference(vector<int>& nums) {
        vector<int> res(nums.size(), 0);
        int leftSum = 0;
        int rightSum = 0;
        for(int i = 0; i < nums.size(); i++)
        {
            res[i] = abs(res[i] - leftSum);
            leftSum += nums[i];
            res[nums.size() - i - 1] = abs(res[nums.size() - i - 1] - rightSum);
            rightSum += nums[nums.size() - i - 1];
        }
        return res;
    }
};

File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.py
class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        res = [0 for _ in range(len(nums))]
        leftSum = 0
        rightSum = 0
        for i in range(len(nums)):
            res[i] = abs(res[i] - leftSum)
            leftSum += nums[i]
            res[-i - 1] = abs(res[-i - 1] - rightSum)
            rightSum += nums[-i - 1]
        return res